packages/shared/src/components/modals/bookmarks/BookmarkFolderSponsorModal.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { ReactElement, useEffect } from 'react';
+import { Modal, ModalProps } from '../common/Modal';
+import { ModalClose } from '../common/ModalClose';
+import { ButtonIconPosition, ButtonVariant } from '../../buttons/common';
+import { Image } from '../../image/Image';
+import { bookmarkFolderSoonImage } from '../../../lib/image';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../../typography/Typography';
+import {
+  useActions,
+  usePlusSubscription,
+  useViewSize,
+  ViewSize,
+} from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+import { Button } from '../../buttons/Button';
+import { ArrowIcon } from '../../icons';
+import { plusUrl } from '../../../lib/constants';
+import { LogEvent, TargetId } from '../../../lib/log';
+import Link from '../../utilities/Link';
+
+type BookmarkFolderSponsorModalProps = Omit<ModalProps, 'children'>;
+
+export const BookmarkFolderSponsorModal = (
+  props: BookmarkFolderSponsorModalProps,
+): ReactElement => {
+  const { onRequestClose } = props;
+  const { checkHasCompleted, isActionsFetched, completeAction } = useActions();
+  const { logSubscriptionEvent } = usePlusSubscription();
+  const isTablet = useViewSize(ViewSize.Tablet);
+
+  useEffect(() => {
+    if (
+      isActionsFetched &&
+      !checkHasCompleted(ActionType.CreateBookmarkFolder)
+    ) {
+      completeAction(ActionType.CreateBookmarkFolder);
+    }
+  }, [checkHasCompleted, completeAction, isActionsFetched]);
```

Comment:
I don't think you need a useEffect here?
Either just fire it all the time or simple inline check it's not set yet.

Reply:
Just didn't want to check for every render, I think is safer limiting it. Wdyt?